app/scripts/lib/app-start.js

```diff
@@ -105,47 +101,57 @@ define(function (require, exports, module) {
 
     initializeConfig () {
       return this._configLoader.fetch()
-                    .then(_.bind(this.useConfig, this))
-                    .then(_.bind(this.initializeAble, this))
-                    .then(_.bind(this.initializeErrorMetrics, this))
-                    .then(_.bind(this.initializeOAuthClient, this))
-                    // both the metrics and router depend on the language
-                    // fetched from config.
-                    .then(_.bind(this.initializeRelier, this))
-                    // metrics depends on the relier.
-                    .then(_.bind(this.initializeMetrics, this))
-                    // iframe channel depends on the relier and metrics
-                    .then(_.bind(this.initializeIframeChannel, this))
-                    // fxaClient depends on the relier and
-                    // inter tab communication.
-                    .then(_.bind(this.initializeFxaClient, this))
-                    // depends on iframeChannel and interTabChannel
-                    .then(_.bind(this.initializeNotifier, this))
-                    // assertionLibrary depends on fxaClient
-                    .then(_.bind(this.initializeAssertionLibrary, this))
-                    // profileClient depends on fxaClient and assertionLibrary
-                    .then(_.bind(this.initializeProfileClient, this))
-                    // marketingEmailClient depends on config
-                    .then(_.bind(this.initializeMarketingEmailClient, this))
-                    // user depends on the profileClient, oAuthClient,
-                    // assertionLibrary and notifier.
-                    .then(_.bind(this.initializeUser, this))
-                    // broker relies on the user, relier, fxaClient,
-                    // assertionLibrary, and metrics
-                    .then(_.bind(this.initializeAuthenticationBroker, this))
-                    // depends on the authentication broker
-                    .then(_.bind(this.initializeHeightObserver, this))
-                    // storage format upgrades depend on user
-                    .then(_.bind(this.upgradeStorageFormats, this))
-
-                    // depends on nothing
-                    .then(_.bind(this.initializeFormPrefill, this))
-                    // depends on notifier, metrics
-                    .then(_.bind(this.initializeRefreshObserver, this))
-                    // router depends on all of the above
-                    .then(_.bind(this.initializeRouter, this))
-                    // appView depends on the router
-                    .then(_.bind(this.initializeAppView, this));
+        .then(_.bind(this.useConfig, this));
+    },
+
+    initializeDeps () {
+      // neither config nor l10n fetch from the network and can be loaded in parallel.
+      return p.all([
+        this.initializeConfig(),
+        this.initializeL10n(),
```

Comment:
I think this *should* work since `initializeConfig` no longer creates any dependencies. initializeConfig doesn't depend on the translator, and the translator doesn't depend on config. this._translator is guaranteed to be available by time `this.initializeAble` and below are called. [service-name.js]() is created by the Sync relier, which is further down the chain than `initializeTranslator`, it seems like `this._translator` should be available?